### DISCUSSION TITLE
Init Grapes on preview mode

### DISCUSSION THREAD

**Discussion by julia-alberici** [GrapesJS/grapesjs#5005] [Q&A] [answered]
I'm trying to implement a "readonly" page where i load a grapes page using loadProjectData() and do editor.Commands.run('preview'); when the editor start but seems like i'm missing something since it doesnt work, it's like the preview mode is running before something that is needed run first but a don't know what it is (if i use a setTimeOut of 1sec it works). I'm using react on the project, can somebody tell me if there's a way to start the editor with data (loaded by loadProjectData()) already on preview mode? I aprecciate any help!

Update: Here's a demo to demonstrate what im talking about https://codesandbox.io/s/grapesjs-react-components-forked-y7s9ne?file=/src/index.js i'm doing editor.Commands.run("preview"); right after the editor.setComponents() in this case,   you can see that a can't really edit the page but the visually behaviours persist (hover, select, toolbar)

**Comment by artf** [comment]
The proper way would be to rely on [`editor.onReady`](https://grapesjs.com/docs/api/editor.html#onready)

**Reply by julia-alberici** [reply]
Hey @artf, thanks for answer me! I tried it on my demo and you were right, it worked! But when i tried on my code it didn't, so after some debug i discover that i have two pieces of code that make this happen 1st on the init function:
 `canvas: {
    scripts: ["https://cdn.tailwindcss.com"]
  },`
and the 2nd is that im using `loadProjectData()`
i have no idea why this happens but if i remove this pieces of code everything works as expected. Can u tell me if this is a bug or am i just doing something wrong?  Here's the updated demo: https://codesandbox.io/s/grapesjs-react-components-forked-h3j496?file=/src/index.js

**Comment by julia-alberici** [comment, marked as answer]
I just solved the problem, when using `loadProjectData()` we have to wait for the HTML be loaded to set preview mode, so `editor.on('update', () => { editor.runCommand('preview'); });` do the trick :)
